src/components/FavoritingTileButton/FavoritingTileButton.tsx

```diff
@@ -0,0 +1,76 @@
+import { ComponentProps, forwardRef } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+import { Text, TextVariants } from '../Text';
+
+// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
+export interface FavoritingTileButtonProps extends ComponentProps<'button'> {
+  /**
+   * Text for when the object is not in the list and the user wants to add it
+   */
+  actionAddText?: string;
+  /**
+   * Text for when the object is in the list and the user wants to remove it
+   */
+  actionRemoveText?: string;
+  /**
+   *  If the selected lot is in the list of listTitle, controls border style and action text displayed on tile
+   */
+  isLotInList: boolean;
+  /**
+   *  List title shown on tile
+   */
+  listTitle: string;
```

Comment:
should these be ReactNodes instead in case we get a future requirement to add icons or something?